src/app/docs/components/search/search.tsx

```diff
@@ -0,0 +1,124 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames/bind';
+import Link from 'next/link.js';
+import MOCKED_SEARCH from '../../mocked_search';
+import { Language } from '@/shared/types';
+
+import styles from './search.module.scss';
+
+const cx = classNames.bind(styles);
+
+const isRunningInDev = process.env.NODE_ENV === 'development';
+
+const translations = {
+  en: {
+    search: {
+      placeholder: 'Search docs...',
+      noResults: 'No results found for "{{query}}". Please try a different search.',
+    },
+  },
+  ru: {
+    search: {
+      placeholder: 'Поиск по документации...',
+      noResults: 'Результатов для "{{query}}" не найдено. Пожалуйста, попробуйте другой запрос.',
+    },
+  },
+};
+
+type SearchProps = {
+  lang: Language;
+};
+
+export default function Search({ lang }: SearchProps) {
+  const [query, setQuery] = useState('');
+  const [results, setResults] = useState<PagefindSearchResult[]>([]);
+
+  useEffect(() => {
+    async function loadPagefind() {
+      if (!isRunningInDev) {
+        window.pagefind = await import(
+          // pagefind.js generated after build
+          /* webpackIgnore: true */ `/_next/static/pagefind/${lang}/pagefind.js`
+        );
+
+        await window.pagefind.options!({ baseUrl: `/_next/static/pagefind/${lang}` });
+      } else {
+        window.pagefind = { search: async () => ({ results: MOCKED_SEARCH }) as unknown as PagefindSearchResults };
+      }
+    }
+
+    loadPagefind();
+  }, [lang]);
```

Comment:
can you please explain this hacking using comments

Reply:
it will throw an error without it:
![image](https://github.com/user-attachments/assets/4e8f1254-37d5-4c5f-9111-bde17529ab01)

you can find more info here: https://www.petemillspaugh.com/nextjs-search-with-pagefind#dynamically-import-search-bundle

Reply:
i mean let's add this explanation link in the code